### DISCUSSION TITLE
Spinnaker Capture error in Bonsai

### DISCUSSION THREAD

**Discussion by AshhadQazi** [bonsai-rx/bonsai#1819] [Q&A]
We are trying to set up the Blackfly S USB3 camera (Model: BFS-U3-16S2M-CS: 1.6 MP, 226 FPS, Sony IMX273, Mono) in Bonsai using a typical Bonsai layout, either with the VideoCapture or the Spinnaker Capture version. We encountered the following errors and could not get it to work in Bonsai.

We plan to try downgrading to an older version of Spinnaker to see if that resolves the issue.

![image](https://github.com/bonsai-rx/bonsai/assets/170653199/038a0ded-9e1a-4a06-80fa-c5108ac5a39a)

![image](https://github.com/bonsai-rx/bonsai/assets/170653199/8411a974-627b-4235-9b17-219fe31e138b)

![image](https://github.com/bonsai-rx/bonsai/assets/170653199/3a259f8d-c560-4b21-81d3-7d7a9a224646)

I have read some older dicussions of people who had the same issue, and they were recommended to:
1)  downgrade the Spinnaker driver to the right version
2)  recompile the Bonsai.Spinnaker package targeting the newer driver

My first guess is that our Spinnaker SDK version is too new? However which one should I try to downgrade to?
link to all the download versions: https://www.flir.com/support-center/iis/machine-vision/downloads/spinnaker-sdk-download/spinnaker-sdk--download-files/

thank you.

**Comment by RoboDoig** [comment]
What version of the SDK are you currently using? I believe the Bonsai package is currently compatible with version 1.29.0.5 which should be available under 'Previous Versions' on the Spinnaker downloads page.

**Comment by bruno-f-cruz** [comment]
You need to expose the output of SpinnakerCapture to apply any transform on top of the image (https://github.com/orgs/bonsai-rx/discussions/1747#discussioncomment-9295770)

You might also need to install the version that @RoboDoig.
